pkg/rockset.go

```diff
@@ -1,380 +1,356 @@
-    if err != nil {
-      return nil, err
-    }
-    times = append(times, t)
-  }
-
-  // add the time dimension
-  frame.Fields = append(frame.Fields, data.NewField("time", labels, times))
-  // add values
-  if len(values) > 0 {
-    frame.Fields = append(frame.Fields, data.NewField(valueField, labels, values))
-  } else if len(strValues) > 0 {
-    frame.Fields = append(frame.Fields, data.NewField(valueField, labels, strValues))
-  }
-  return frame, nil
+	frame := data.NewFrame(valueField)
+
+	var times []time.Time
+	var values []float64
+	var strValues []string
+
+	var labels map[string]string
+	// empty label means there is no label column and labels should use the zero value, which is nil
+	if labelColumn != "" {
+		labels = map[string]string{labelColumn: label}
+	}
+
+	// iterate over the rows
+	for _, m := range qr.Results {
+		if labelColumn != "" {
+			if l, found := m[labelColumn]; found && l != label {
+				// skip rows which doesn't match the label
+				continue
+			}
+		}
+
+		// the value might not be present in every row
+		v, found := m[valueField]
+		if !found {
+			continue
+		}
+
+		// TODO: this is a bit naïve and could be improved, as the rows can be of different types
+		f, ok := v.(float64)
+		if ok {
+			values = append(values, f)
+		}
+		if !ok {
+			// TODO: is there a way to send warnings back to the Grafana UI?
+			g, strOk := v.(string)
+			if strOk {
+				strValues = append(strValues, g)
+			}
+			if !strOk {
+				log.DefaultLogger.Error("could not cast to float64 or string", "column", valueField, "value", v)
+				continue
+			}
+		}
+
+		// TODO: time conversion could be cached as it is parsed each call to makeFrame
+		t, err := parseTime(m, timeField)
+		if err != nil {
+			return nil, err
+		}
+		times = append(times, t)
+	}
+
+	// add the time dimension
+	frame.Fields = append(frame.Fields, data.NewField("time", labels, times))
+	// add values
+	if len(values) > 0 {
+		frame.Fields = append(frame.Fields, data.NewField(valueField, labels, values))
+	} else if len(strValues) > 0 {
+		frame.Fields = append(frame.Fields, data.NewField(valueField, labels, strValues))
+	}
+	return frame, nil
 }
 
 func parseTime(fields map[string]interface{}, key string) (time.Time, error) {
-  ifc, ok := fields[key]
-  if !ok {
-    // TODO include a list of available columns
-    return time.Time{}, fmt.Errorf("could not find column %s in query response", key)
-  }
-
-  k, ok := ifc.(string)
-  if !ok {
-    return time.Time{}, fmt.Errorf("could not cast %s to string", key)
-  }
-
-  t, err := time.Parse(time.RFC3339Nano, k)
-  if err != nil {
-    return time.Time{}, fmt.Errorf("failed to convert %s to time: %w", k, err)
-  }
-
-  return t, nil
+	ifc, ok := fields[key]
+	if !ok {
+		// TODO include a list of available columns
+		return time.Time{}, fmt.Errorf("could not find column %s in query response", key)
+	}
+
+	k, ok := ifc.(string)
+	if !ok {
+		return time.Time{}, fmt.Errorf("could not cast %s to string", key)
```

Comment:
this can provide more information
```suggestion
		return time.Time{}, fmt.Errorf("could not cast %s (%v) %T to string", key, ifc, ifc)
```